tasks/common/apiHelper.ts

```diff
@@ -108,6 +108,47 @@ export function deleteOldPackages(submissionPackages: any, numberOfPackagesToKee
     });
 }
 
+/**
+ * Compares two version strings, returns a positive integer if X is greated than Y,
+ * negative if Y is greater than X, 0 if equal.
+ * @param x The version string to compare in the form of X[.X]* | X : number
+ * @param y The version string to compare in the form of X[.X]* | X : number
+ * Examples:
+ * 2 > 1.5.1 > 1.5.0.45
+ * 1.7 == 1.7.0.0
+ */
+function compareVersions(x: string, y: string): number {
+    var i = 0;
+    var xParts = x.split('.');
+    var yParts = y.split('.');
+    var minLengthToCompare = Math.min(xParts.length, yParts.length);
+
+    // Compare common parts
+    for (i = 0; i < minLengthToCompare; i++) {
+        var diff = parseInt(xParts[i], 10) - parseInt(yParts[i], 10);
+        if (diff) {
+            return diff;
+        }
+    }
+
+    if (xParts.length == yParts.length) {
+        return 0;
+    }
+
+    // Min set of common segments is equal, check for extra segments greater than zero
+    var longerVersion = minLengthToCompare == xParts.length ? yParts : xParts;
```

Comment:
consider simplifying by pushing 0's to the lesser array, xParts or yParts, to make them equal in length